src/tests/authentication/test_token.py

```diff
@@ -0,0 +1,221 @@
+import json
+import uuid
+from datetime import datetime, timedelta
+from time import sleep
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+from django.test import TestCase
+
+from chaosinventory.authentication.models import Token, User
+
+
+def get_db_token(key) -> Token:
+    return Token.objects.get(key=key)
+
+
+def create_user(**kwargs) -> User:
+    password = None
+    if 'password' in kwargs:
+        password = kwargs.pop('password')
+
+    default_user_data = {
+        'username': uuid.uuid4(),
+        'first_name': 'Demo',
+        'last_name': 'User',
+        'email': 'user@example.org',
+    }
+    user: User = get_user_model().objects.create(**{**default_user_data, **kwargs})
+
+    if password is not None:
+        user.set_password(password)
+        user.save()
+
+    return user
+
+
+class TokenTestCase(TestCase):
+    def setUp(self) -> None:
+        self.user_password: str = 'password'
+        self.user = create_user(password=self.user_password)
+
+    def obtain_token(self, username: str = None, password: str = None, application: str = None,
```

Comment:
I don't know if this is working for a long time, because obtaining tokens with authentication is planned as a temporary feature.

Maybe better we should create a token in the database and after that test the other endpoints (I don't think the `/api/authentication/token/credentials` needs to be tested).

Reply:
As discussed, we'll leave it in because it won't hurt but for only temporary features we shouldn't bother implementing test.